src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexBoyerMoore.cs

```diff
@@ -40,16 +40,17 @@ public RegexBoyerMoore(string pattern, bool caseInsensitive, bool rightToLeft, C
             // We're doing this for your own protection. (Really, for speed.)
             Debug.Assert(pattern.Length != 0, "RegexBoyerMoore called with an empty string. This is bad for perf");
 
-            // We do the ToLower character by character for consistency.  With surrogate chars, doing
-            // a ToLower on the entire string could actually change the surrogate pair.  This is more correct
-            // linguistically, but since Regex doesn't support surrogates, it's more important to be
-            // consistent.
             if (caseInsensitive)
             {
-                StringBuilder sb = StringBuilderCache.Acquire(pattern.Length);
-                for (int i = 0; i < pattern.Length; i++)
-                    sb.Append(culture.TextInfo.ToLower(pattern[i]));
-                pattern = StringBuilderCache.GetStringAndRelease(sb);
+                pattern = string.Create(pattern.Length, (pattern, culture), (span, state) =>
+                {
+                    // We do the ToLower character by character for consistency.  With surrogate chars, doing
+                    // a ToLower on the entire string could actually change the surrogate pair.  This is more correct
+                    // linguistically, but since Regex doesn't support surrogates, it's more important to be
+                    // consistent.
+                    for (int i = 0; i < state.pattern.Length; i++)
+                        span[i] = state.culture.TextInfo.ToLower(state.pattern[i]);
```

Comment:
Cache state._culture.TextInfo in a local in front of the loop?

Reply:
Does that matter? I see that it's lazily cached per culture.

Reply:
Presumably the local can potentially be enregistered or at least doesn't need 2 indirections to get.

Reply:
`CultureInfo.TextInfo` is virtual property. You would be saving a virtual call per each loop iteration.